Source/WebCore/platform/graphics/PixelBuffer.cpp

```diff
@@ -38,30 +38,58 @@ bool PixelBuffer::supportedPixelFormat(PixelFormat pixelFormat)
     switch (pixelFormat) {
     case PixelFormat::RGBA8:
     case PixelFormat::BGRA8:
+#if HAVE(HDR_SUPPORT)
+    case PixelFormat::RGBA16F:
+#endif
         return true;
 
     case PixelFormat::BGRX8:
     case PixelFormat::RGB10:
     case PixelFormat::RGB10A8:
-#if HAVE(HDR_SUPPORT)
-    case PixelFormat::RGBA16F:
-#endif
         return false;
     }
 
     ASSERT_NOT_REACHED();
     return false;
 }
 
-CheckedUint32 PixelBuffer::computeBufferSize(PixelFormat pixelFormat, const IntSize& size)
+static CheckedUint32 mustFitInInt32(CheckedUint32 uint32)
+{
+    if (!uint32.hasOverflowed() && uint32.value() > uint32_t(std::numeric_limits<int32_t>::max()))
+        uint32.overflowed();
+    return uint32;
```

Comment:
I see you are moving existing code, but you can use `isInBounds` from CheckedArithmetic.h to make this a little more concise:

```
if (!isInBounds<int32_t>(uint32.value()))
    uint32.overflowed();
return uint32;
```

or, and the compiler can probably optimize this to be equivalent to the above:

```
    return CheckedInt32(uint32);
```